runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -149,6 +150,16 @@
     @Getter
     volatile CompletableFuture<Void> connectionFuture;
 
+    /**
+     * A future to a periodic tasks that sends pings
+     */
+    private ScheduledFuture<?> keepAliveTask;
```

Comment:
make this volatile.

Reply:
Actually, sonarcloud will complain that you use volatile on (rind of) async onject.
You could use asyncObject instead, like: `private final Object asyncObject = new Object()`

Reply:
A channel is single threaded, the events that change the value of keepAliveTask will be called by the same thread. Let me know if this is not how the netty model works.

Reply:
Yes, that makes sense. I was comparing it with `channelFuture` but that's used by multiple threads.